test/e2e/storage/volume_expand.go

```diff
@@ -0,0 +1,196 @@
+		Expect(pvc).NotTo(BeNil())
+
+		pvcSize := pvc.Spec.Resources.Requests[v1.ResourceStorage]
+		if pvcSize.Cmp(newSize) != 0 {
+			framework.Failf("Error updating pvc size %s", pvc.Name)
+		}
+
+		By("Waiting for cloudprovider resize to finish")
+		err = waitForControllerVolumeResize(pvc, c)
+		Expect(err).NotTo(HaveOccurred(), "While waiting for pvc resize to finish")
+
+		By("Checking for conditions on pvc")
+		pvc, err = c.CoreV1().PersistentVolumeClaims(ns).Get(pvc.Name, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred(), "While fetching pvc after controller resize")
+
+		inProgressConditions := pvc.Status.Conditions
+		Expect(len(inProgressConditions)).To(Equal(1), "pvc must have resize condition")
+		Expect(inProgressConditions[0].Type).To(Equal(v1.PersistentVolumeClaimResizing), "pvc must have resizing condition")
+
+		By("Delete the previously created pod")
+		err = framework.DeletePodWithWait(f, c, pod)
+		Expect(err).NotTo(HaveOccurred(), "while deleting pod for resizing")
+
+		By("Create new pod with same volume")
+		pod2, err := framework.CreatePod(c, ns, nil, pvcClaims, false, "")
+		Expect(err).NotTo(HaveOccurred(), "while recreating pod for resizing")
+		defer func() {
+			err = framework.DeletePodWithWait(f, c, pod2)
+			Expect(err).NotTo(HaveOccurred(), "while cleaning up pod before exiting resizing test")
+		}()
+
+		By("Waiting for file system resize to finish")
+		pvc, err = waitForFSResize(pvc, c)
+		Expect(err).NotTo(HaveOccurred(), "while waiting for fs resize to finish")
+
+		pvcConditions := pvc.Status.Conditions
+		Expect(len(pvcConditions)).To(Equal(0), "pvc should not have conditions")
+	})
+})
+
+func createResizableStorageClass(t storageClassTest, ns string, suffix string, c clientset.Interface) (*storage.StorageClass, error) {
+	stKlass := newStorageClass(t, ns, suffix)
+	allowExpansion := true
+	stKlass.AllowVolumeExpansion = &allowExpansion
+
+	var err error
+	stKlass, err = c.StorageV1().StorageClasses().Create(stKlass)
+	return stKlass, err
+}
+
+func expandPVCSize(origPVC *v1.PersistentVolumeClaim, size resource.Quantity, c clientset.Interface) (*v1.PersistentVolumeClaim, error) {
+	pvcName := origPVC.Name
+	updatedPVC := origPVC.DeepCopy()
+
+	waitErr := wait.PollImmediate(2*time.Second, 30*time.Second, func() (bool, error) {
+		var err error
+		updatedPVC, err = c.CoreV1().PersistentVolumeClaims(origPVC.Namespace).Get(pvcName, metav1.GetOptions{})
+		if err != nil {
+			return false, fmt.Errorf("Error fetching pvc %s for resizing with %v", pvcName, err)
+		}
+
+		updatedPVC.Spec.Resources.Requests[v1.ResourceStorage] = size
+		updatedPVC, err = c.CoreV1().PersistentVolumeClaims(origPVC.Namespace).Update(updatedPVC)
+		if err == nil {
+			return true, nil
+		}
+		framework.Logf("Error updating pvc %s with %v", pvcName, err)
+		return false, nil
+	})
+	return updatedPVC, waitErr
+}
+
+func waitForControllerVolumeResize(pvc *v1.PersistentVolumeClaim, c clientset.Interface) error {
```

Comment:
is there value in exporting this func to other test packages?

Reply:
other test packages? It is in small case, how can this be exported?

Reply:
I was thinking that `waitForControllerVolumeResize` and the other wait func might be generally useful e2e functions. If so they should be public and moved to a common utility file (eg. framework/util.go or storage/pv-util.go)...

Reply:
Oh okay, I misunderstood. May be we can do this later as scope of resizing covers more section of e2e tests.